File: lg5-spring-kafka/lg5-spring-kafka-producer/src/main/java/com/lg5/spring/kafka/producer/service/KafkaProducer.java
package com.lg5.spring.kafka.producer.service;

import org.apache.avro.specific.SpecificRecordBase;
import org.springframework.kafka.support.SendResult;

import java.io.Serializable;
import java.util.function.BiConsumer;

public interface KafkaProducer<K extends Serializable, V extends SpecificRecordBase> {

    void send(String topicName, K key, V message, BiConsumer<SendResult<K, V>, Throwable> callback);

}

File: lg5-spring-kafka/lg5-spring-kafka-publisher/src/main/kotlin/com/lg5/spring/kafka/publisher/service/KafkaProducer.kt
package com.lg5.spring.kafka.publisher.service

import org.apache.avro.specific.SpecificRecordBase
import org.springframework.kafka.support.SendResult
import java.io.Serializable
import java.util.function.BiConsumer


interface KafkaProducer<K : Serializable, V : SpecificRecordBase> {
    fun send(topicName: String, key: K, message: V, callback: BiConsumer<SendResult<K, V>, Throwable>)
}